src/annotator/host.coffee

```diff
@@ -4,6 +4,24 @@ Guest = require('./guest')
 
 module.exports = class Host extends Guest
   constructor: (element, options) ->
+
+    # Some options' values are not JSON-stringifiable (e.g. JavaScript
+    # functions) and will be omitted when the options are JSON-stringified.
+    # Add a JSON-stringifiable option for each of these so that the sidebar can
+    # at least know whether the callback functions were provided or not.
+    if options.services?[0]
+      service = options.services[0]
+      if service.onLoginRequest
+        service.onLoginRequestProvided = true
+      if service.onLogoutRequest
+        service.onLogoutRequestProvided = true
+      if service.onSignupRequest
+        service.onSignupRequestProvided = true
+      if service.onProfileRequest
+        service.onProfileRequestProvided = true
+      if service.onHelpRequest
+        service.onHelpRequestProvided = true
```

Comment:
Mapping functions to booleans is fine, but the problem with the way it is done here is that it mutates the input options data structure, which could cause code outside this module to break.

An alternative method which avoids the problem would be to use the `replacer` parameter of `JSON.stringify()` at the point where the config is serialized into a string for passing into the sidebar app.

```js
JSON.stringify({ aNumber: 42, aString: "hello", aFunction: () => {}, anObject: {} },
               (key, value) => typeof value === 'function' ? true : value)
// '{"aNumber":42,"aString":"hello","aFunction":true,"anObject":{}}'
```